dialog/file.go

```diff
@@ -292,6 +298,12 @@ func showFile(file *FileDialog) *fileDialog {
 		(fileIconSize+fileTextSize)+theme.Padding()*4))
 
 	d.win = widget.NewModalPopUp(ui, file.parent.Canvas())
+	showHiddenShortcut := &desktop.CustomShortcut{KeyName: fyne.KeyH, Modifier: desktop.ControlModifier}
+	d.win.Canvas.AddShortcut(showHiddenShortcut, func(shortcut fyne.Shortcut) {
```

Comment:
If you are adding a keyboard shortcut into the canvas when this shows it needs to be removed when it is hidden.
Otherwise we are either leaking the UI to satisfy the callback, or it will crash as the UI is destroyed.

Reply:
I was unable to find a method to remove shortcuts so I added one to the canvas interface. Not sure if this change would be a desired approach. For this ticket it feels like scope creep.

Reply:
Good point. I'd say we can't really use canvas.AddShortcut for temporal shortcuts, so maye move that to another PR along with the code that adds the ability to remove them?

Reply:
Would you mind evaluating a bit on how you envision temporal shortcuts to work? I'm picking up the code base as I go, so I doubt I would have a well versed initial approach to its implementation.

Shortcuts like copy/paste feel like they would always be available through the system so these feel like maybe "persistent" shortcuts.

Temporary shortcuts like the hide/show hidden files shortcut would only available for a single dialog. Would the canvas not be the ideal place to manage these?

Reply:
Shortcuts for inputs (like Entry) are registered on the entry, so come in to effect when that is focused.
In this case we have not got a focused input so the canvas is the right place - but we are only shown for a short while so the same shortcut should be removed once we hide. The dialog does not have a separate canvas you see.
I'll check out your PR now, I hope this is what you assumed.